### PR TITLE
G UI

### PR DESCRIPTION
**Pull request by Cristian-Cruz7**

### CODE REVIEW

File: App/__init__.py [hunk 1]
```diff
@@ -1,7 +1,13 @@
 from flask import Flask
 
+#APP_INSTANCE = False
 
-def create_app():
-    app = Flask(__name__)
+def get_app_instance():
+    # if APP_INSTANCE:
```
Comment (Cruzlex7) [inline App/__init__.py]: is this commented block in use?

**Review by Cruzlex7 — CHANGES_REQUESTED** [review]
Please don't upload cache files. Add it to the .gitignore file

File: App/routes/base_endpoint.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+def api():
+   return render_template('Home.html')
+
+@app.route('/api/otraApp')
+def api_otraApp():
+   return render_template('otraApp.html')
+
+@app.route('/api/calc')
+def api_calc():
+   return render_template('calculadora.html')
```
Comment (Cruzlex7) [inline App/routes/base_endpoint.py]: Please add and end the white line.

File: App/routes/operations.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+from flask import jsonify
+from main import app
+
+@app.route("/api/calc/sumar/<num1>/<num2>")
+def api_calc_sumar(num1, num2):
+   resultado = int(num1) + int(num2)
+   #return render_template('Suma.html', resultado = resultado, num1 = num1, num2 = num2)
```
Comment (Cruzlex7) [inline App/routes/operations.py]: remove commented code, comments should be used to add documentation to the app.

File: App/routes/operations.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+from flask import jsonify
+from main import app
+
+@app.route("/api/calc/sumar/<num1>/<num2>")
```
Comment (Cruzlex7) [inline App/routes/operations.py:4]: Please add comments to describe what this function does (docs)

File: App/static/css/estilos.css [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+.calc {
+  margin: 0 auto;
+  padding: 20px;
+  background-color: rgba(63, 80, 241, 0.4);
+  width: 300px;
+  height: 300px;
+  border-radius: 10px;
+  padding: 20px;
+  user-select: none;
```
Comment (Cruzlex7) [inline App/static/css/estilos.css:31]: add prefix for internet explorer, and mozila support
```suggestion
  user-select: none;
  
    -webkit-touch-callout: none; /* iOS Safari */
    -webkit-user-select: none; /* Safari */
     -khtml-user-select: none; /* Konqueror HTML */
       -moz-user-select: none; /* Old versions of Firefox */
        -ms-user-select: none; /* Internet Explorer/Edge */
            user-select: none; /* Non-prefixed version, currently
                                  supported by Chrome, Edge, Opera and Firefox */
```

File: App/static/css/estilos.css [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+.calc td button {
+  display: block;
+  background-color: white;
+  width: 60px;
+  height: 60px;
+  font-size: 20px;
+}
+
+#resultado {
```
Comment (Cruzlex7) [inline App/static/css/estilos.css]: Is a good practice to use a class selector, not Id one. 
```suggestion
.resultado {
```

File: App/static/css/estilos.css [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  display: block;
+  width: 30px;
+  height: 30px;
+  font-size: 25px;
+  text-decoration: none;
+  text-align: center;
+  font-weight: bold;
+  border: solid 1px black;
+  background-color: rgba(63, 80, 241, 0.2);
+}
```
Comment (Cruzlex7) [inline App/static/css/estilos.css]: please, add end white line for all

File: App/templates/calculadora.html [hunk 8]
```diff
@@ -1,14 +1,49 @@
...
+  <head>
+    <link
+      rel="stylesheet"
+      href="{{ url_for('static', filename ='css/estilos.css')}}"
+    />
+  </head>
+
+  <body onload="init();">
+    <div>
+      <a href="{{ url_for('api') }}" class="regresar"><</a>
```
Comment (Cruzlex7) [inline App/templates/calculadora.html]: `class` attribute should be before, then `href `

File: App/templates/calculadora.html [hunk 9]
```diff
@@ -1,14 +1,49 @@
...
+        </tr>
+        <tr>
+          <td><button id="uno">1</button></td>
+          <td><button id="dos">2</button></td>
+          <td><button id="tres">3</button></td>
+          <td><button id="cero">0</button></td>
+        </tr>
+      </table>
+    </div>
+    <script src="../static/funciones.js"></script>
```
Comment (Cruzlex7) [inline App/templates/calculadora.html]: Nice!, Is a good practice module and separate logic from HTML. God Job

File: main.py [hunk 10]
```diff
@@ -1,42 +1,6 @@
...
-
-@app.route("/api/calc/Division/8/2")
-def api_calc_Division_8_2():
-    return render_template('Div.html')
+from app import get_app_instance
 
+app = get_app_instance()
 
+from app.routes.base_endpoint import *
+from app.routes.operations import *
```
Comment (Cruzlex7) [inline main.py]: Add end white line.

File: App/__init__.py (hunk 1 above)
Comment (Cristian-Cruz7) [reply App/__init__.py]: Done

File: App/routes/base_endpoint.py (hunk 2 above)
Comment (Cristian-Cruz7) [reply App/routes/base_endpoint.py]: done

File: App/routes/operations.py (hunk 3 above)
Comment (Cristian-Cruz7) [reply App/routes/operations.py]: fixed

File: App/static/css/estilos.css (hunk 5 above)
Comment (Cristian-Cruz7) [reply App/static/css/estilos.css:31]: fixed

File: App/static/css/estilos.css (hunk 6 above)
Comment (Cristian-Cruz7) [reply App/static/css/estilos.css:37]: done

File: App/static/css/estilos.css (hunk 7 above)
Comment (Cristian-Cruz7) [reply App/static/css/estilos.css]: fixed

File: App/templates/calculadora.html (hunk 8 above)
Comment (Cristian-Cruz7) [reply App/templates/calculadora.html]: fixed

File: App/templates/calculadora.html (hunk 9 above)
Comment (Cristian-Cruz7) [reply App/templates/calculadora.html]: thanks.

File: App/routes/operations.py (hunk 4 above)
Comment (Cristian-Cruz7) [reply App/routes/operations.py:4]: Done, Comments added to the function

File: App/static/css/estilos.css (hunk 6 above)
Comment (Cristian-Cruz7) [reply App/static/css/estilos.css]: done

**Pull request closed without merging**